extensions/mssql/src/errorDiagnostics/errorDiagnosticsService.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ISqlOpsFeature, SqlOpsDataClient, SqlOpsFeature } from 'dataprotocol-client';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import * as contracts from '../contracts';
+import { AppContext } from '../appContext';
+import { ServerCapabilities, ClientCapabilities, RPCMessageType } from 'vscode-languageclient';
+import { Disposable } from 'vscode';
+
+export const diagnosticsId = 'azurediagnostics'
+export const serviceName = 'AzureDiagnostics';
+
+export class ErrorDiagnosticsService extends SqlOpsFeature<any> {
+	private static readonly messagesTypes: RPCMessageType[] = [
+		contracts.DiagnosticsRequest.type,
+	];
+
+	public static asFeature(context: AppContext): ISqlOpsFeature {
+		return class extends ErrorDiagnosticsService {
+			constructor(client: SqlOpsDataClient) {
+				super(context, client);
+			}
+
+			override fillClientCapabilities(capabilities: ClientCapabilities): void { }
+
+			override initialize(): void {
+				this.register(this.messages, {
+					id: UUID.generateUuid(),
+					registerOptions: undefined
+				});
+			}
+
+			protected override registerProvider(options: any): Disposable {
+				const client = this._client;
+
+				let handleErrorCode = (errorCode: number, errorMessage: string): Thenable<azdata.diagnostics.ErrorDiagnosticsResponse> => {
+					const params: contracts.ErrorDiagnosticsParameters = { errorCode, errorMessage };
+					try {
+						return client.sendRequest(contracts.DiagnosticsRequest.type, params);
```

Comment:
If removing STS, you can still maintain this service, and client references can be replaced with direct errorcode comparisons.